### ISSUE TITLE
0.31 bugs list

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry#1755]
- [x] BLOCKER: don't delete source files when removing old claims (on claim update with new file) - ```updating claim to stream generated from 8ff278da, deleted previous stream ```
- [x] Blocker for Franklin?: handle download timeout cleanly:
```2019-02-04 19:39:27,500 INFO     lbrynet.stream.stream_manager:301: returned from get lbry://drevo-blademaster-keyboard-of-the-future#930ac211e90d84413ab96cb205f1879285dcb39b
2019-02-04 19:39:27,501 ERROR    lbrynet.extras.daemon.Daemon:492: error handling api request
Traceback (most recent call last):
 File "lbrynet/extras/daemon/Daemon.py", line 489, in _process_rpc_call
 File "lbrynet/extras/daemon/Daemon.py", line 1579, in jsonrpc_get
lbrynet.error.DownloadSDTimeout: Failed to download sd blob drevo-blademaster-keyboard-of-the-future#930ac211e90d84413ab96cb205f1879285dcb39b within timeout
```
- [x] blocker?: failed to decrypt blob spam on startup / resuming downloads where blobs renaming <> 0 - status is set to running, completed is false (but all blobs downloaded). Can we reset all these statuses? 

- [x] Blocker?: if I delete both blobs + downloaded file, can't redownload the file. Get just hangs for 30 seconds, no logs.  This will also happen if you download sd blob only (fail on data blobs) + try to redownload later. 

- [x] blocker?: blob server/announcer and file manager don't start until the wallet is fully synced

- [x] unknown length / race condition ? This is a local PC on my network. 
```019-01-29 10:03:23,106 ERROR    lbrynet.blob_exchange.client:56: error downloading blob from 24.184.108.248:3333: unknown blob length
2019-01-29 10:03:23,107 ERROR    lbrynet.blob_exchange.client:136: race happened downloading from 24.184.108.248:3333
```

- [x] long delay between detecting ip and loading file manager, when UPNP fails I think ([see slack](https://lbryians.slack.com/archives/C81K0BM9A/p1548376754493000))

- [x] peer errors- not enough values to unpack  : ```2019-01-29 09:22:08,678 WARNING  lbrynet.dht.protocol.protocol:435: error raised handling b'store' request from 66.18.203.205:4444 - <class 'ValueError'>(not enough values to unpack (expected 5, got 1))```

- [x] announcement not working after publish/ download. manual blob announce not working either - says true, but doesn't announce.
- [x] reflector should show success after all blobs reflected. 
- [x] blocker: after publishing, reflector doesn't show when file was completed reflecting - only shows the descriptor being sent.
- [x] `lbrynet claim list one` fails (needs lbryumx server update). 
- [x] ```asyncio.base_futures.InvalidStateError: invalid state```
- [x] skipping reflector component doesn't actually skip it - still connects. 
- [x] some logging like ``` torba.client.basedatabase:164: connecting to database: c:\lbrynew3\lbrynet.sqlite``` is only shown in console, not logged to the logfile.
- [x] Blocker? : files that failed to download (had 1 head blob, no sd blob) won't re-startup / redownload. Not sure how it got into this state yet - seemed to happen when downloads failed during heavy announcement activity. 
- [x] crashing with ```ValueError: too many file descriptors in select()``` on one install when a download was in "running" status.  Seemed to stop crashing after the file was deleted, but it came back. 
- [x] missing file count in status (used to say xxx files loaded)
- [x] blob announcement not working properly? See sd blobs + head blobs in pending state - only 1 blob announced after a download. Shouldn't it be sd blob + head blob?. Also see lots of old blobs that haven't been anounced, and have invalid dates like: 
![image](https://user-images.githubusercontent.com/8120721/52180573-20d6f400-27b6-11e9-991c-dcdbae31d2d8.png)

- [x] uploaded streams to reflector? Is it really uploading? I'm seeing this a bunch.
```2019-02-02 08:01:03,153 INFO     lbrynet.stream.stream_manager:168: uploaded 4 streams to reflector
2019-02-02 08:06:03,155 INFO     lbrynet.stream.stream_manager:168: uploaded 3 streams to reflector
2019-02-02 08:10:55,998 WARNING  lbrynet.dht.protocol.protocol:487: known dht protocol backwards compatibility error with 173.88.175.206:4444 (lbrynet v0.19.1)
2019-02-02 08:11:03,165 INFO     lbrynet.stream.stream_manager:168: uploaded 3 streams to reflector
2019-02-02 08:16:03,176 INFO     lbrynet.stream.stream_manager:168: uploaded 3 streams to reflector
2019-02-02 08:21:03,168 INFO     lbrynet.stream.stream_manager:168: uploaded 3 streams to reflector
2019-02-02 08:26:03,175 INFO     lbrynet.stream.stream_manager:168: uploaded 3 streams to reflector
```

- [x] if file is deleted during a download, we should delete the downloaded file by default (why would I want to keep it?

- [x] ```Task was destroyed but it is pending!
task: <Task pending coro=<ManagedStream.upload_to_reflector() running at c:\users\thoma\lbry\lbrynet\stream\managed_stream.py:202> wait_for=<Task pending coro=<Queue.get() running at c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\queues.py:159> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x0B1D5B10>()]> cb=[<TaskWakeupMethWrapper object at 0x06A92570>()]>>
Task was destroyed but it is pending!
task: <Task pending coro=<Queue.get() running at c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\queues.py:159> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x0B1D5B10>()]> cb=[<TaskWakeupMethWrapper object at 0x06A92570>()]>```
- [x] BLOCKER: Key payment error: 

```2019-01-31 21:54:30,375 ERROR    lbrynet.extras.daemon.Daemon:479: error handling api request
Traceback (most recent call last):
  File "c:\users\thoma\lbry\lbrynet\extras\daemon\Daemon.py", line 476, in _process_rpc_call
    result = await result
  File "c:\users\thoma\lbry\lbrynet\extras\daemon\Daemon.py", line 1586, in jsonrpc_get
    self.dht_node, resolved, file_name, timeout, fee_amount, fee_address
  File "c:\users\thoma\lbry\lbrynet\stream\stream_manager.py", line 229, in download_stream_from_claim
    await self.wallet.send_amount_to_address(lbc_to_dewies(str(fee_amount)), fee_address.encode('latin1'))
AttributeError: 'bytes' object has no attribute 'encode'
```

- [x] node id mismatch?
```2019-01-31 21:51:43,330 ERROR    lbrynet.dht.protocol.protocol:491: node id mismatch in reply
2019-01-31 21:51:43,330 ERROR    lbrynet.dht.protocol.protocol:491: node id mismatch in reply
Task exception was never retrieved
future: <Task finished coro=<IterativeFinder._send_probe() done, defined at c:\users\thoma\lbry\lbrynet\dht\protocol\iterative_find.py:162> exception=TimeoutError(b'V\x85\x1bn\x02\xda\x92\xf6~\xa6\xd2\x91\xa9\xa8\xfdco\x8f\t\x07\xb4?(\x003\x83\x1d\xd2\x00\xef\x1e\xa6\xd22\x10\xb4+\xd5w8"\x7f\xf8_\x7f\xe5\x10\xf6')>
Traceback (most recent call last):
  File "c:\users\thoma\lbry\lbrynet\dht\protocol\iterative_find.py", line 164, in _send_probe
    response = await self.send_probe(peer)
  File "c:\users\thoma\lbry\lbrynet\dht\protocol\iterative_find.py", line 270, in send_probe
    response = await self.protocol.get_rpc_peer(peer).find_node(self.key)
  File "c:\users\thoma\lbry\lbrynet\dht\protocol\protocol.py", line 166, in find_node
    self.peer, RequestDatagram.make_find_node(self.protocol.node_id, key)
  File "c:\users\thoma\lbry\lbrynet\dht\protocol\protocol.py", line 524, in send_request
    response = await asyncio.wait_for(response_fut, self.rpc_timeout)
  File "c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\tasks.py", line 412, in wait_for
    return fut.result()
TimeoutError: b'V\x85\x1bn\x02\xda\x92\xf6~\xa6\xd2\x91\xa9\xa8\xfdco\x8f\t\x07\xb4?(\x003\x83\x1d\xd2\x00\xef\x1e\xa6\xd22\x10\xb4+\xd5w8"\x7f\xf8_\x7f\xe5\x10\xf6'
```

- [x] BLOCKER?: claim updates (with metadata or new file) fail to re-download/update - 
```already have matching stream for lbry://testupdate1#01f429c603f085326c4e122ce929a01b289db3ca```. Previous behavior: metadata updates would say "already have file" and update the file_list entry with new data. File updates would download the new file and create a 2nd file_list entry. Neither happens now, it just stalls. 

- [x] BLOCKER: file list entry not updated after failed/duplicate downloads (notice file name is without a suffix, but has one in the download log)
```download_path": "C:\\Users\\IEUser\\Downloads\\survival-cutty-things-condor.mp4```
```
2019-01-29 17:49:26,816 INFO     lbrynet.stream.downloader:219: downloaded stream b15b1234e79082be980e5e93e663a1f81f3a79014f9d4d25533c12d83e051fc5307c6269803c4fd0a9ab37da0f284fd2 -> C:\Users\IEUser\Downloads\survival-cutty-things-condor_3.mp4
```

- [x] BLOCKER: file path ends up blank after file is deleted from download directory - 
```"download_directory": "C:\\Users\\IEUser\\Downloads",
    "download_path": null,
    "file_name": "a-brief-history-of-pokemon_1.mp4"
```

- [x] BLOCKER: can't rebuild file if deleted from downloads folder, calling get returns
```2019-01-29 18:00:19,680 INFO     lbrynet.extras.daemon.Daemon:1581: already have matching stream for salman-khan-reaction-on-total-dhamaal-2```

- [x] BLOCKER: deleting a file during download does not cleanup file in downloads folder. Can't delete because it's used by lbrynet.exe.  file list does not return when this happens. 

- [x] BLOCKER: can't update file without file_list entry: ```Exception: no previous stream to update```

- [x] BLOCKER: download timeout, peers/fixed avaialble:
```9-01-29 16:14:24,837 INFO     lbrynet.stream.stream_manager:213: get lbry://survival-cutty-things-condor-knife-tool#e56e8287ffcb5ec2d646ee2441658914ba45e939
2019-01-29 16:14:24,848 INFO     lbrynet.stream.downloader:216: download and decrypt stream
2019-01-29 16:14:24,849 INFO     lbrynet.stream.downloader:84: downloader idle...
2019-01-29 16:14:26,841 INFO     lbrynet.stream.downloader:150: added 1 new peers
2019-01-29 16:14:26,842 INFO     lbrynet.stream.downloader:163: adding fixed peer 18.213.255.15:5567
2019-01-29 16:14:27,044 INFO     lbrynet.blob_exchange.client:104: downloaded b15b1234 from 18.213.255.15:5567
2019-01-29 16:14:27,049 INFO     lbrynet.stream.stream_manager:174: got descriptor b15b1234e79082be980e5e93e663a1f81f3a79014f9d4d25533c12d83e051fc5307c6269803c4fd0a9ab37da0f284fd2 for survival-cutty-things-condor-knife-tool
2019-01-29 16:14:27,050 INFO     lbrynet.stream.downloader:172: add head blob hash to peer search
2019-01-29 16:14:27,227 INFO     lbrynet.stream.downloader:150: added 3 new peers
2019-01-29 16:14:27,304 INFO     lbrynet.stream.downloader:150: added 2 new peers
2019-01-29 16:14:27,387 INFO     lbrynet.stream.downloader:150: added 3 new peers
2019-01-29 16:14:27,579 WARNING  lbrynet.blob_exchange.client:79: blob not in availability response from 85.17.24.157:4444
2019-01-29 16:14:29,870 INFO     lbrynet.stream.downloader:150: added 10 new peers
2019-01-29 16:14:35,878 INFO     lbrynet.stream.downloader:84: downloader idle...
2019-01-29 16:14:53,630 WARNING  lbrynet.dht.protocol.protocol:487: known dht protocol backwards compatibility error with 199.115.205.245:4444 (lbrynet v0.19.1)
2019-01-29 16:14:54,840 INFO     lbrynet.stream.downloader:211: stopped downloader
2019-01-29 16:14:54,841 INFO     lbrynet.stream.stream_manager:241: returned from get lbry://survival-cutty-things-condor-knife-tool#e56e8287ffcb5ec2d646ee2441658914ba45e939
2019-01-29 16:14:54,842 ERROR    lbrynet.extras.daemon.Daemon:478: error handling api request
Traceback (most recent call last):
  File "lbrynet\extras\daemon\Daemon.py", line 475, in _process_rpc_call
  File "lbrynet\extras\daemon\Daemon.py", line 1589, in jsonrpc_get
lbrynet.error.DownloadSDTimeout: Failed to download sd blob b15b1234e79082be980e5e93e663a1f81f3a79014f9d4d25533c12d83e051fc5307c6269803c4fd0a9ab37da0f284fd2 within timeout
```

- [x] uncaught exception when client disconnects
```2019-01-29 19:46:18,040 INFO     lbrynet.blob_exchange.server:60: send 93bc8a13 to 194.158.76.133:54079
Task exception was never retrieved
future: <Task finished coro=<BlobServerProtocol.handle_request() done, defined at c:\users\thoma\lbry\lbrynet\blob_exchange\server.py:35> exception=ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None)>
Traceback (most recent call last):
  File "c:\users\thoma\lbry\lbrynet\blob_exchange\server.py", line 61, in handle_request
    sent = await blob.sendfile(self)
  File "c:\users\thoma\lbry\lbrynet\blob\blob_file.py", line 133, in sendfile
    return await self.loop.sendfile(writer.transport, handle)
  File "c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\base_events.py", line 1050, in sendfile
    offset, count)
  File "c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\base_events.py", line 1073, in _sendfile_fallback
    await proto.drain()
  File "c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\base_events.py", line 181, in drain
    await fut
  File "c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\selector_events.py", line 891, in _write_ready
    n = self._sock.send(self._buffer)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
```

- [x] Created separate issue: fresh install startup failure - we need to make sure all directories/files are created.  It works after creating lbry/lbrynet ```FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\IEUser\\AppData\\Local\\lbry\\lbrynet\\lbrynet.log'```

- [x] BLOCKER: previously failed downloads try to resume with new version? ```lbrynet.stream.stream_manager:97: resuming 5 downloads```. On restart, it tries to resume 2 files every time, starting the downloads, and then stalling - should it not start to download the same blobs it failed on, but only new ones?  
- Jack explained that it doens't actually download them, but may log to decode them. 

- [x] new timeout error? 
```2019-01-29 16:51:01,301 INFO     lbrynet.dht.protocol.iterative_find:331: max results
Task exception was never retrieved
future: <Task finished coro=<IterativeFinder._send_probe() done, defined at lbrynet\dht\protocol\iterative_find.py:160> exception=TimeoutError(b'V\x85\x1bn\x02\xda\x92\xf6~\xa6\xd2\x91\xa9\xa8\xfdco\x8f\t\x07\xb4?(\x003\x83\x1d\xd2\x00\xef\x1e\xa6\xd22\x10\xb4+\xd5w8"\x7f\xf8_\x7f\xe5\x10\xf6')>
Traceback (most recent call last):
  File "lbrynet\dht\protocol\iterative_find.py", line 162, in _send_probe
  File "lbrynet\dht\protocol\iterative_find.py", line 295, in send_probe
  File "lbrynet\dht\protocol\protocol.py", line 166, in find_node
  File "lbrynet\dht\protocol\protocol.py", line 524, in send_request
  File "asyncio\tasks.py", line 416, in wait_for
```

- [x] verbose logging? 
```2019-01-29 09:13:24,797 INFO     lbrynet.dht.node:235: peer search: [<lbrynet.dht.peer.KademliaPeer object at 0x082BB8B0>, <lbrynet.dht.peer.KademliaPeer object at 0x082BB530>, <lbrynet.dht.peer.KademliaPeer object at 0x082BB390>, <lbrynet.dht.peer.KademliaPeer object at 0x082BB830>, <lbrynet.dht.peer.KademliaPeer object at 0x082BB610>, <lbrynet.dht.peer.KademliaPeer object at 0x082BB110>, <lbrynet.dht.peer.KademliaPeer object at 0x109EEBF0>, <lbrynet.dht.peer.KademliaPeer object at 0x082BB870>]```
- [x] BLOCKER: errors when stopping daemon ```2019-01-29 09:15:57,780 INFO     lbrynet.extras.daemon.exchange_rate_manager:216: Stopping exchange rate manager
Task was destroyed but it is pending!
task: <Task pending coro=<StreamDownloader._download() running at c:\users\thoma\lbry\lbrynet\stream\downloader.py:217> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x0F7F0750>()]>>
2019-01-29 09:15:58,033 INFO     lbrynet.stream.downloader:211: stopped downloader
Task was destroyed but it is pending!
task: <Task pending coro=<StreamDownloader._download() running at c:\users\thoma\lbry\lbrynet\stream\downloader.py:217> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x07335C50>()]>>
2019-01-29 09:15:58,036 INFO     lbrynet.stream.downloader:211: stopped downloader
Task was destroyed but it is pending!
task: <Task pending coro=<Event.wait() running at c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\locks.py:293> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x0F7F0770>()]> cb=[_wait.<locals>._on_completion() at c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\tasks.py:436]>
Task was destroyed but it is pending!
task: <Task pending coro=<Event.wait() running at c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\locks.py:293> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x0998B350>()]> cb=[_wait.<locals>._on_completion() at c:\users\thoma\appdata\local\programs\python\python37-32\lib\asyncio\tasks.py:436]>```

- [x] there are file_delete issues per @jackrobison .

- [x] BLOCKER: download stalled on lbry://a-brief-history-of-pokemon-clones#eb52cea650dd51e09228725f5a74e64d87a64ec6
```2019-01-26 12:24:47,198 INFO     lbrynet.stream.downloader:82: adding download probes for 0 peers to 2 already active
2019-01-26 12:24:48,281 WARNING  lbrynet.dht.protocol.protocol:487: known dht protocol backwards compatibility error with 71.231.158.82:4444 (lbrynet v0.19.1)
2019-01-26 12:24:51,190 WARNING  lbrynet.dht.protocol.protocol:487: known dht protocol backwards compatibility error with 71.231.158.82:4444 (lbrynet v0.19.1)
2019-01-26 12:24:59,476 INFO     lbrynet.stream.downloader:64: drop peer 24.184.108.248:3333
2019-01-26 12:24:59,477 INFO     lbrynet.stream.downloader:69: keep peer 149.115.96.252:3333
2019-01-26 12:24:59,478 INFO     lbrynet.stream.downloader:84: downloader idle...
```

- [x] assert error after failed download (after deleting file)
```Traceback (most recent call last):
  File "c:\users\thoma\lbry\lbrynet\extras\daemon\Daemon.py", line 475, in _process_rpc_call
    result = await result
  File "c:\users\thoma\lbry\lbrynet\extras\daemon\Daemon.py", line 1591, in jsonrpc_get
    raise DownloadSDTimeout(resolved['value']['stream']['source']['source'])
lbrynet.error.DownloadSDTimeout: Failed to download sd blob afec0dbb9a1e12e0bbde9e7fd48f5004b596e6497ff48b2358a6bac82bd172fd7a6c511a8d2483ed9e2af804b4e908bf within timeout
Task exception was never retrieved
future: <Task finished coro=<StreamDownloader._download() done, defined at c:\users\thoma\lbry\lbrynet\stream\downloader.py:217> exception=AssertionError()>
Traceback (most recent call last):
  File "c:\users\thoma\lbry\lbrynet\stream\downloader.py", line 221, in _download
    await self.assemble_decrypted_stream(self.output_dir, self.output_file_name)
  File "c:\users\thoma\lbry\lbrynet\stream\assembler.py", line 71, in assemble_decrypted_stream
    self.sd_blob)
  File "c:\users\thoma\lbry\lbrynet\stream\descriptor.py", line 122, in from_stream_descriptor_blob
    return await loop.run_in_executor(None, lambda: cls._from_stream_descriptor_blob(loop, blob_dir, blob))
  File "c:\users\thoma\appdata\local\programs\python\python37-32\lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "c:\users\thoma\lbry\lbrynet\stream\descriptor.py", line 122, in <lambda>
    return await loop.run_in_executor(None, lambda: cls._from_stream_descriptor_blob(loop, blob_dir, blob))
  File "c:\users\thoma\lbry\lbrynet\stream\descriptor.py", line 95, in _from_stream_descriptor_blob
    assert os.path.isfile(blob.file_path)
```

- [x] file (lbry://patreon-account-deletion#1b110f51cbe87ebc75c5b37aac9d91c2972491dd) downloaded with a blank file_name, can't play it in app. 
- [x] time to stream script doesn't start daemon, [fails after 2nd uri](https://lbryians.slack.com/archives/C81K0BM9A/p1548427447514700): ```raise JSONRPCException(data['error'])
lbrynet.extras.daemon.client.JSONRPCException```

- [x] download_directory was changed to download_dir - We need to convert this for users or support download_directory as well.

- [x] strange that ```lbrynet file_list``` is no longer valid, but is valid as a json command. 

- [x] stopped downloader [log spam]
(https://lbryians.slack.com/archives/C81K0BM9A/p1548371923479700)

- [x] not seeing torba info in logs - missing blocks, transaction info

- [x] BLOCKER: publishing broke new / old sdk startup, [see slack](https://lbryians.slack.com/archives/C81K0BM9A/p1548376021490700). 
- [x] running lbrnet / lbrynet --help doesn't tell you what startup arguments are possible
- [x] ssl errors when connecting to loggly/exchange rate manager data (https://github.com/lbryio/lbry/pull/1213/files) 
- [x] seeing ```lbrynet.dht.blob_announcer:27: 43067 blobs to announce``` - is this because announce head blob is set to 0? 0.30.4 shows no blobs to announce. 
- [x] configs default to wrong values, toggle bug
- [x] files not being reflected or announced
- [x] BLOCKER: publish process does not create file_list entry, also seeing: ```    'written_bytes': None if not full_path else self.downloader.written_bytes or os.stat(full_path).st_size,
AttributeError: 'NoneType' object has no attribute 'written_bytes'```
- [x] BLOCKER: `written_bytes` goes to zero upon restarting the daemon
- [x] when upnp completes or fails, get a never awaitd error ```c:\users\thoma\lbry\lbrynet\extras\daemon\Components.py:617: RuntimeWarning: coroutine 'Manager.send_upnp_setup_success_fail' was never awaited
  self.component_manager.analytics_manager.send_upnp_setup_success_fail(success, await self.get_status())```

```2019-01-29 15:48:59,522 INFO     lbrynet.extras.daemon.Components:601: detecting external ip
2019-01-29 15:48:59,537 INFO     lbrynet.extras.daemon.exchange_rate_manager:211: Starting exchange rate manager
2019-01-29 15:48:59,631 INFO     lbrynet.extras.daemon.Components:226: remote height: 510938, local height: 0
2019-01-29 15:49:02,756 INFO     lbrynet.extras.daemon.Components:188: fetched headers from s3 (s3 height: 510936), now verifying integrity after download.
2019-01-29 15:49:15,116 WARNING  lbrynet.extras.daemon.Components:537: upnp discovery failed: M-SEARCH for 10.0.2.2:1900 timed out
2019-01-29 15:49:15,116 WARNING  lbrynet.extras.daemon.Components:551: unable to get external ip from UPnP, checking lbry.io fallback
2019-01-29 15:49:15,193 ERROR    lbrynet.extras.daemon.Components:616: failed to setup upnp
2019-01-29 15:49:15,694 INFO     lbrynet.extras.daemon.Components:377: start the dht
2019-01-29 15:49:15,694 INFO     lbrynet.extras.daemon.Components:401: Started the dht
```

- [x] [timeout errors](https://github.com/lbryio/lbry/issues/1779) ```lbrynet\dht\protocol\protocol.py:132> exception=TimeoutError()>```

- [x] claim_list broken in some cases, i.e. `claim_list one`:   "message": "string indices must be integers" /   "message": "(-32603, 'internal server error')"
- [ ] ```lbrynet API listening on TCP ::1:5279``` - used to show localhost. this works sometimes, not others: ```TCP 127.0.0.1:5279```
- [x]  integrity error on file_delete
- [x]  Pretty sure this is related to sending blobs - random ServerRequestHandler error/traceback.
```
019-01-09 10:03:18,430 WARNING  lbrynet.p2p.server.ServerRequestHandler:160: An error occurred handling a client request. Error message: 'dict' object is not callable
2019-01-09 10:03:18,431 WARNING  lbrynet.p2p.server.ServerRequestHandler:160: An error occurred handling a client request. Error message: 'Deferred' object is not callable
2019-01-09 10:03:18,431 WARNING  lbrynet.p2p.server.ServerRequestHandler:126: An error occurred handling a request. Error: 'dict' object is not callable
Unhandled error in Deferred:
2019-01-09 10:03:18,431 CRITICAL twisted:154: Unhandled error in Deferred:

2019-01-09 10:03:18,431 CRITICAL twisted:154:
Traceback (most recent call last):
  File "twisted\internet\defer.py", line 151, in maybeDeferred
TypeError: 'Deferred' object is not callable
Unhandled error in Deferred:
2019-01-09 10:03:18,432 CRITICAL twisted:154: Unhandled error in Deferred:
```

- [x] if a daemon is already running, starting a new one freezes up / doesn't quit. ctrl-c won't quit sometimes either. 

- [ ] empty list returned? lbrynet blob_reflect 3addd2c9e1c0f8e038947de8a64e1b5d4697690c7448f1b74ba7704193bc55c03b875369d04b8656c3de887f67b50363
```
[]
log shows: lbrynet.extras.reflector.client.client:77: Reflector has all blobs for f7bd08071c1a740c
```
- [x] ~should say what the argument is: lbrynet blob_announce~ 
```
  "message": "single argument must be specified
```
- [x] ~coroutine 'Daemon.jsonrpc_peer_list' was never awaited~
```
2019-01-10 17:03:17,556 INFO     lbrynet.dht.node:227: Attempting to join the DHT network, 10 contacts known so far
2019-01-10 17:03:28,502 INFO     lbrynet.dht.node:263: Joined the dht
2019-01-10 17:03:50,820 INFO     torba.client.baseledger:133: lbc_mainnet: added BlockHeightEvent(height=500605, change=1) header blocks, final height 500605
C:\Users\thoma\AppData\Local\Temp\_MEI277122\lbrynet\extras\daemon\Daemon.py:961: RuntimeWarning: coroutine 'Daemon.jsonrpc_peer_list' was never awaited
2019-01-10 17:06:09,160 INFO     lbrynet.extras.daemon.HashAnnouncer:60: Announcing 47 blobs
```
- [x] this tries to download the blob for some reason (weird connecting logs, see below) and craps out: lbrynet blob_availability 3addd2c9e1c0f8e038947de8a64e1b5d4697690c7448f1b74ba7704193bc55c03b875369d04b8656c3de887f67b50363
```
 builtins.AttributeError: 'NoneType' object has no attribute 'execute'
builtins.AttributeError: 'NoneType' object has no attribute 'rollback'
---
2019-01-10 21:12:34,643 INFO     torba.client.basedatabase:164: connecting to database: :memory:
2019-01-10 21:12:34,645 INFO     torba.client.basedatabase:164: connecting to database: :memory:
2019-01-10 21:12:34,645 INFO     torba.client.basedatabase:164: connecting to database: :memory:
2019-01-10 21:12:34,942 INFO     lbrynet.p2p.client.BlobRequester:415: Offered rate 0.000000/mb accepted by 85.17.24.157

```
- [x] uncaught exception in peer response? 
```Unhandled error in Deferred:
2019-01-18 12:42:11,755 CRITICAL twisted:154: Unhandled error in Deferred:

2019-01-18 12:42:11,755 CRITICAL twisted:154:
Traceback (most recent call last):
  File "twisted\internet\task.py", line 517, in _oneWorkUnit
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "twisted\internet\defer.py", line 1418, in _inlineCallbacks
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "twisted\internet\defer.py", line 654, in _runCallbacks
  File "lbrynet\extras\wallet\manager.py", line 579, in _request_failed
Exception: A peer failed to send a valid public key response. Error: , peer: 24.184.108.248:3333
```
- [ ] dropped api request exception not handled
```019-01-18 15:25:47,520 ERROR    aiohttp.server:355: Unhandled exception
Traceback (most recent call last):
  File "site-packages\aiohttp\web_protocol.py", line 447, in start
  File "site-packages\aiohttp\web_response.py", line 353, in prepare
  File "site-packages\aiohttp\web_response.py", line 667, in _start
  File "site-packages\aiohttp\web_response.py", line 410, in _start
  File "site-packages\aiohttp\http_writer.py", line 112, in write_headers
  File "site-packages\aiohttp\http_writer.py", line 67, in _write
ConnectionResetError: Cannot write to closing transport
```
- [x] handle disconnected state better? 
```Unhandled error in Deferred:
2019-01-21 16:53:02,074 CRITICAL twisted:154: Unhandled error in Deferred:

2019-01-21 16:53:02,074 CRITICAL twisted:154:
Traceback (most recent call last):
  File "twisted\internet\defer.py", line 1416, in _inlineCallbacks
  File "twisted\python\failure.py", line 491, in throwExceptionIntoGenerator
  File "lbrynet\log_support.py", line 31, in _emit
  File "twisted\internet\defer.py", line 654, in _runCallbacks
  File "twisted\internet\endpoints.py", line 975, in startConnectionAttempts
twisted.internet.error.DNSLookupError: DNS lookup failed: no results for hostname lookup: logs-01.loggly.com.
2019-01-21 16:53:02,069 ERROR    lbrynet.extras.daemon.analytics:265: Encountered an exception while POSTing to https://api.segment.io/v1/track:
Traceback (most recent call last):
  File "site-packages\aiohttp\connector.py", line 955, in _create_direct_connection
  File "site-packages\aiohttp\connector.py", line 825, in _resolve_host
  File "site-packages\aiohttp\resolver.py", line 30, in resolve
  File "asyncio\base_events.py", line 778, in getaddrinfo
  File "concurrent\futures\thread.py", line 57, in run
  File "socket.py", line 748, in getaddrinfo
socket.gaierror: [Errno 11001] getaddrinfo failed

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "lbrynet\extras\daemon\analytics.py", line 262, in _post
  File "site-packages\aiohttp\client.py", line 1043, in __aenter__
  File "site-packages\aiohttp\client.py", line 476, in _request
  File "site-packages\aiohttp\connector.py", line 522, in connect
  File "site-packages\aiohttp\connector.py", line 854, in _create_connection
  File "site-packages\aiohttp\connector.py", line 959, in _create_direct_connection
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host api.segment.io:443 ssl:None [getaddrinfo failed]
```

![shipit](https://spee.ch/shipit)

**Comment by tzarebczan** [comment]
Opened new issues for outstanding bugs.
